### PR TITLE
Bump eslint version

### PR DESCRIPTION
**Pull request by TylerHorth**
I don't believe there are any breaking changes. I tested esify quickly, it seems to work fine, and fixes more errors 👍 

@lemonmade @GoodForOneFare

### CODE REVIEW

File: packages/shopify-codemod/transforms/global-assignment-to-default-export.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default function globalAssignmentToDefaultExport({source}, {jscodeshift:
   }
 
   function isDefaultExportMemberAssignment(defaultExportName, assignmentTargetString) {
-    return new RegExp(`^${defaultExportName}\..+`).test(assignmentTargetString);
+    return new RegExp(`^${defaultExportName}\\..+`).test(assignmentTargetString);
```
Comment (TylerHorth) [inline packages/shopify-codemod/transforms/global-assignment-to-default-export.js]: The new eslint caught this 👌 
Need to escape the escape in order for the regex to match the literal period.

Comment (GoodForOneFare) [inline packages/shopify-codemod/transforms/global-assignment-to-default-export.js]: This looks useful, but should probably be in a separate PR.  It's easier to update CHANGELOG if we can just scan through commit messages.

**Comment by GoodForOneFare** [comment]
Awesome! 🎆

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-15.0.3
\ No newline at end of file
+15.1.0
```
Comment (lemonmade) [inline VERSION]: Don't edit this file directly, it will be edited when we publish the new version

File: packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js [hunk 3]
```diff
@@ -21,6 +21,8 @@ module.exports = {
   'eol-last': 'warn',
   // Disallow space between function identifier and application
   'func-call-spacing': 'warn',
+  // Require function names to match the name of the variable or property to which they are assigned
+  'func-name-matching': 'warn',
```
Comment (TylerHorth) [inline packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js]: @lemonmade How do you feel about this rule? http://eslint.org/docs/rules/func-name-matching

Comment (lemonmade) [inline packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js]: I know we also have a rule that prevents shadowing, if you do

```js
const foo = function foo() {}
```

With the new rules does it warn that the function `foo` is shadowing? If not, then 👍

Comment (TylerHorth) [inline packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js]: So it doesn't warn about shadowing, but it does complain about `func-style` since we have it set to declaration only. 

I suppose it couldn't hurt to have this rule as well, there may be some places that it covers where `func-style` doesn't.

Comment (lemonmade) [inline packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js]: :+1:

File: packages/chai-jscodeshift/index.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ var config = {
 module.exports = function chaiJSCodeShift(options) {
   var finalConfig = merge({}, config, options || {});
 
-  // eslint-disable-next-line no-shadow, func-style
+  // eslint-disable-next-line no-shadow, func-style, func-name-matching
   var chaiHelper = function chaiJSCodeShift(chaiOptions) {
```
Comment (lemonmade) [inline packages/chai-jscodeshift/index.js]: Rename the function here to `chaiHelper` and we should be able to ditch the `no-shadow` and `func-name-matching` eslint disables.

**Comment by GoodForOneFare** [comment]
🚢
